include/openenclave/sgx/seal.h

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/**
+ * @file
+ *
+ * This file defines SGX specific constants and structures for sealing APIs.
+ *
+ * Only SGX specific definitions should go here, while TEE generic definitions
+ * should go in bits/seal.h
+ *
+ */
+#ifndef _OE_SGX_SEAL_H
+#define _OE_SGX_SEAL_H
+
+#include <openenclave/bits/sgx/sgxtypes.h>
+#include <openenclave/seal.h>
+
+OE_EXTERNC_BEGIN
+
+/**
+ * SGX specific seal settings.
+ */
+enum oe_seal_setting_type_sgx_t
+{
+    OE_SEAL_SETTING_SGX_KEYNAME = OE_SEAL_SETTING_PLUGIN_DEFINED,
+    OE_SEAL_SETTING_SGX_ISVSVN,
+    OE_SEAL_SETTING_SGX_CET_ATTRIBUTES_MASK,
+    OE_SEAL_SETTING_SGX_CPUSVN,
+    OE_SEAL_SETTING_SGX_FLAGSMASK,
+    OE_SEAL_SETTING_SGX_XFRMMASK,
+    OE_SEAL_SETTING_SGX_MISCMASK,
+    OE_SEAL_SETTING_SGX_CONFIGSVN,
+
+    /**
+     * Lower bound of plug-in defined setting types.
+     */
+    OE_SEAL_SETTING_SGX_PLUGIN_DEFINED =
+        OE_SEAL_SETTING_PLUGIN_DEFINED + OE_SEAL_SETTING_PLUGIN_DEFINED / 2
+};
+
+/**
+ * Initialize a \c oe_seal_setting_t structure to specify \c
+ * KEYREQUEST.KEYNAME.
+ *
+ * @param w 16-bit value to be assigned to \c KEYREQUEST.KEYNAME.
```

Comment:
```suggestion
 * @param[in] w 16-bit value to be assigned to \c KEYREQUEST.KEYNAME.
```

Add [in] style annotation in this file (here and below)